helpscout/hsds-react
# React 16!

## React 16!!!

![](https://media.giphy.com/media/E8uiHr1z6shMY/giphy.gif)

This very special update bumps all the important bits of HSDS React to the latest/greatest, including:

* React 16
* Webpack 4
* Babel 7
* Storybook 4
* Enzyme 3
* Typescript 3

### How does one update to React 16?

Very simply, actually. There were only a handful of things I needed to tweak. Most of them were warnings vs. breaking failures. The only component that needed some React 16 compatibility work was `Portal`. I've enhanced it to allow for native `ReactDOM.createPortal`, if the user wishes (and if ReactDOM@16 is present).

### What's up with all the changes then?

Enzyme... ENZYME!!!!

(Enzyme 3 to be exact)

Enzyme 3 changed how their APIs worked, especially when it comes to things like `mount().find` or `.node`/`.instance()`, things like that.

The majority of this work is from updating Enzyme 2->3 tests.

### How do I test?

* Pull it down
* `npm install`
* `npm start` (Check to see if Storybook works)
* `npm test` (Check to see if Jest works)
* `npm run build` (Check to see if it builds)

That's about it :). I mean.. you can totally scrub through all the Enzyme test updates if you want... 😊 

### Does it still work with 15?

Yes! Since HSDS: React is still backwards compatible with React 15 👍 

I've manually tested it with:

Beacon2 and HS-App in Chrome, Firefox, Safari, and Edge.

Things are working as expected
